docs/features/nullable-metadata.md

```diff
@@ -0,0 +1,189 @@
+
+`NullableContextAttribute` is valid in metadata at the module level and at type and method declarations.
+The `byte` value represents the implicit `NullableAttribute` value for type references within that scope
+that do not have an explicit `NullableAttribute` and would not otherwise be represented by an empty `byte[]`.
+The nearest `NullableContextAttribute` in the metadata hierarchy applies.
+If there are no `NullableContextAttribute` attributes in the hierarchy,
+missing `NullableAttribute` attributes are treated as `NullableAttribute(0)`.
+
+The attribute is not inherited.
+
+The C#8 compiler uses the following algorithm to determine which `NullableAttribute` and
+`NullableContextAttribute` attributes to emit.
+First, `NullableAttribute` attributes are generated at each type reference and type parameter definition by:
+calculating the `byte[]`, skipping empty `byte[]`, and collapsing `byte[]` to single `byte` where possible.
+Then at each level in metadata hierarchy starting at methods:
+The compiler finds the most common single `byte` value across all the `NullableAttribute` attributes at that level
+and any `NullableContextAttribute` attributes on immediate children.
+If there are no single `byte` values, there are no changes.
+Otherwise, a `NullableContext(value)` attribute is created at that level where `value` is most common
+value (preferring `0` over `1` and preferring `1` over `2`), and all `NullableAttribute` and `NullableContextAttribute` attributes with that value are removed.
+That iterative process continues up to the module level.
+If the common value at the module level is a value other than `0` (the default), a module level `NullableContext(value)` attribute is emitted.
+
+Note that an assembly compiled with C#8 where all reference types are oblivious will have no
+`NullableContextAttribute` and no `NullableAttribute` attributes emitted.
+That is equivalent to a legacy assembly.
+
+### Examples
+```C#
+[NullableContext(2)]
+class Program
+{
+    [Nullable(2)] string s; // string?
+    [Nullable({ 2, 1, 2 }] Dictionary<string, object> d; // Dictionary<string!, object?>?
+    [Nullable(1)] int[] a;           // int[]!
+    int[] b;                         // int[]?
+    [Nullable({ 0, 2 })] object[] c; // object?[]~
+}
+```
+
+## NullableMembersAttribute
+
+`NullableMembersAttribute` can be used to indicate whether nullable attributes were included for `private` or `internal` members.
+
+```C#
+namespace System.Runtime.CompilerServices
+{
+    [System.AttributeUsage(AttributeTargets.Module, AllowMultiple = false)]
+    public sealed class NullableMembersAttribute : Attribute
+    {
+        public readonly bool IncludesPrivateMembers;
+        public readonly bool IncludesInternalMembers;
+
+        public NullableMembersAttribute(bool includesPrivateMembers, bool includesInternalMembers)
```

Comment:
>bool includesPrivateMembers, bool includesInternalMembers [](start = 40, length = 57)

Should this be two unrelated values, or should this be a single value similar metadata import options? Do we want to support a scenario when internals are excluded, but privates are included?